src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -328,6 +328,53 @@ public boolean isAnnotated(TypeMirror e) {
         return !e.getAnnotationMirrors().isEmpty();
     }
 
+    public boolean isRecursivelyAnnotated(TypeMirror e) {
```

Comment:
I'm not sure this needs to be a new public  method in `Utils` since it is only used by `AbstractExcecutableMemberWriter`. It also seems to do more than is necessary for that use case (array, wildcard).

Reply:
Agreed. Moved this to a helper method like `addParameters` in `AbstractExcecutableMemberWriter`. Also changed test so the tests for `Generic2`'s doc group together.